### PR TITLE
class for exceptions

### PR DESCRIPTION
**Pull request by Sergiy-1998**
Define your custom class for exceptions. Show an example how it might be used.

### CODE REVIEW

**Comment by python-teacher** [comment]
Please change the name of this pull request.
It should be self-descriptive. It should be clear from the name what is this pull request about.

File: exception.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"""
+Define your custom class for exceptions. Show an example how it might be used.
+"""
+
+class Calculator:
+    while True:
+        try:
+            a = int(input('a = '))
```
Comment (python-teacher) [inline exception.py]: Please avoid using `input` function.

**Review by python-teacher — CHANGES_REQUESTED** [review]

File: exception.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+            b = int(input('b = '))
+            print("{} / {} = {}".format(a,b,(a/b)))
+        except (ZeroDivisionError, ValueError):
+            print(False)
```
Comment (python-teacher) [inline exception.py]: What is the reason of `print(False)`?

File: exception.py (hunk 1 above)
Comment (Sergiy-1998) [reply exception.py]: take

File: exception.py (hunk 2 above)
Comment (Sergiy-1998) [reply exception.py]: if the wrong type or 0

File: exception.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+class Calculator:
+    def zero(a, b):
+        try:
+            print("{} / {} = {}".format(a, b, (a / b)))
+        except (ZeroDivisionError, ValueError, TypeError):
+            print("{} / {} = {}".format(a, b, False))
+
+
+one = Calculator
```
Comment (python-teacher) [inline exception.py]: What are you doing here?

File: exception.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+Define your custom class for exceptions. Show an example how it might be used.
+"""
+
+
+class Calculator:
+    def zero(a, b):
+        try:
+            print("{} / {} = {}".format(a, b, (a / b)))
+        except (ZeroDivisionError, ValueError, TypeError):
+            print("{} / {} = {}".format(a, b, False))
```
Comment (python-teacher) [inline exception.py]: It's better to print some message about this issue.

File: exception.py (hunk 3 above)
Comment (Sergiy-1998) [reply exception.py]: obviously not what it was necessary)

File: exception.py (hunk 4 above)
Comment (Sergiy-1998) [reply exception.py]: done

File: exception.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+"""
+Define your custom class for exceptions. Show an example how it might be used.
+"""
+
+
+class Calculator:
```
Comment (python-teacher) [inline exception.py]: This class should be inherited from the base class with exceptions.

Comment (Sergiy-1998) [reply exception.py]: done

File: exception.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+"""
+Define your custom class for exceptions. Show an example how it might be used.
+"""
+
+
+class Error(Exception):
+    pass
+
+
+class Type(Error):
```
Comment (python-teacher) [inline exception.py]: It's better to use words as error or exception for custom exception classes.

As we have default classes in Python: TypeError, ValueError, etc.

So, please change names for your classes.

File: exception.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class ZeroDivision(Error):
+    pass
+
+
+def zero(a, b):
+    try:
+        if isinstance(a, str) or isinstance(b, str):
+            raise Type
```
Comment (python-teacher) [inline exception.py]: It's a common way to raise an exception with some message, not just a class.

**Review by python-teacher — CHANGES_REQUESTED** [review]

File: exception.py (hunk 6 above)
Comment (Sergiy-1998) [reply exception.py]: ready, renamed classes

File: exception.py (hunk 7 above)
Comment (Sergiy-1998) [reply exception.py]: I do not understand what to do

Comment (python-teacher) [reply exception.py]: You should understand how to raise an exception properly.

Comment (Sergiy-1998) [reply exception.py]: done

File: exception.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
+"""
+Define your custom class for exceptions. Show an example how it might be used.
+"""
+
+
+class Error(Exception):
+    pass
+
+
+class DivisionDifferentTypes(Error):
```
Comment (python-teacher) [inline exception.py]: This name is not descriptive.

**Comment by python-teacher** [comment]
Class names are not fixed according to previous comment.

File: exception.py (hunk 8 above)
Comment (Sergiy-1998) [reply exception.py]: done

Comment (python-teacher) [reply exception.py]: It's not according to conventions.

Comment (Sergiy-1998) [reply exception.py]: Changed, according to requirements

File: exception.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+class Error(Exception):
+    pass
+
+
+class OtherTypeError(Error):
+    pass
+
+
+class DivideByZero(Error):
```
Comment (python-teacher) [inline exception.py]: Why didn't you rename this class?

File: exception.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+def zero(a, b):
+    try:
+        if isinstance(a, str) or isinstance(b, str):
+            raise OtherTypeError('Invalid type')
+        elif int(b) == 0:
+            raise DivideByZero('Division by zero')
+        else:
+            print("{} / {} = {}".format(a, b, (a / b)))
+    except OtherTypeError as ddt:
```
Comment (python-teacher) [inline exception.py]: what is `ddt`?

File: exception.py (hunk 9 above)
Comment (Sergiy-1998) [reply exception.py]: I thought he was in order, ready

File: exception.py (hunk 10 above)
Comment (Sergiy-1998) [reply exception.py]: abbreviation

File: exception.py (hunk 9 above)
Comment (python-teacher) [reply exception.py]: in which order?

File: exception.py (hunk 10 above)
Comment (python-teacher) [reply exception.py]: of what?

Comment (Sergiy-1998) [reply exception.py]: OtherTypeError it ote:

File: exception.py (hunk 9 above)
Comment (Sergiy-1998) [reply exception.py]: the name of the class, and at the end of the 'error', to make it clear that this is an exception

File: exception.py (hunk 10 above)
Comment (UkrainianEngineer) [reply exception.py]: Do you think it's clear enough for others to understand your abbreviations?

Comment (Sergiy-1998) [reply exception.py]: I think that's the case, because it's used one single time

Comment (Sergiy-1998) [reply exception.py]: If a variable was used several times, then it should be renamed

Comment (python-teacher) [reply exception.py]: each variable should be named with a correct and clear name. It doesn't matter how many times it is used.

Comment (Sergiy-1998) [reply exception.py]: done

**Review by python-teacher — APPROVED** [review]

File: exception.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+def zero(a, b):
+    try:
+        if isinstance(a, str) or isinstance(b, str):
+            raise OtherTypeError('Invalid type')
+        elif int(b) == 0:
+            raise DivideByZeroError('Division by zero')
+        else:
+            print("{} / {} = {}".format(a, b, (a / b)))
+    except OtherTypeError as other_type:
```
Comment (python-teacher) [inline exception.py]: it's better to name it as `other_type_error`, etc. to make sure it's clear what is the content of this variable.

**Comment by python-teacher** [comment]
👍

File: exception.py (hunk 11 above)
Comment (Sergiy-1998) [reply exception.py]: done

**Pull request merged**